### PR TITLE
ioctls: Add open_with_cloexec_at and new_with_path

### PR DESCRIPTION
**Pull request by Lencerf**
### Summary of the PR

These two functions are the same as `open_with_cloexec` and `new` except they allow users to specify the path to the KVM device file.

### Requirements

Before submitting your PR, please make sure you addressed the following
requirements:

- [x] All commits in this PR are signed (with `git commit -s`), and the commit
  message has max 60 characters for the summary and max 75 characters for each
  description line.
- [x] All added/changed functionality has a corresponding unit/integration
  test.
- [x] All added/changed public-facing functionality has entries in the "Upcoming 
  Release" section of CHANGELOG.md (if no such section exists, please create one).
- [x] Any newly added `unsafe` code is properly documented.

### CODE REVIEW

**Review by rminnich — DISMISSED** [review]
very nice, this is going to be very helpful.

**Comment by andreeaflorescu** [comment]
We have a problem with the CI, and we need to update it before we can merge this. The PR looks good to me, and I'll approve it once I get to fix the CI issue (later today).

File: src/ioctls/system.rs
Comment (andreeaflorescu) [inline src/ioctls/system.rs]: You will need to add the "SAFETY:" suffix before the safety comment to make the CI happy. Otherwise this clippy check will fail: https://rust-lang.github.io/rust-clippy/master/#missing_safety_doc.

**Comment by andreeaflorescu** [comment]
@Lencerf I fixed the CI problems I was mentioning, can you please do a rebase and fix the way the unsafe comments are written so that they start with the prefix "SAFETY:"?

File: src/ioctls/system.rs
Comment (Lencerf) [reply src/ioctls/system.rs]: Done. Thank you very much, now All checks have passed.

**Comment by Lencerf** [comment]
Ops, I clicked re-request review and github removed all other reviewers...😕

File: src/ioctls/system.rs
Comment (acatangiu) [inline src/ioctls/system.rs]: Why not test the same `cstr` build as the func?
```suggestion
        let kvm_path = unsafe { CStr::from_bytes_with_nul_unchecked(b"/dev/kvm\0") };
```

**Review by acatangiu — DISMISSED** [review]

File: src/ioctls/system.rs
Comment (Lencerf) [reply src/ioctls/system.rs]: Done.

**Review by andreeaflorescu — APPROVED** [review]

**Review by acatangiu — APPROVED** [review]

**Pull request merged**
